code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -981,20 +981,34 @@
 	if(!istype(apc) || QDELETED(apc) || apc.machine_stat & BROKEN)
 		to_chat(src, span_danger("Hack aborted. The designated APC no longer exists on the power network."))
 		playsound(get_turf(src), 'sound/machines/buzz-two.ogg', 50, TRUE, ignore_walls = FALSE)
-	else if(apc.aidisabled)
+		return
+	if(apc.aidisabled)
 		to_chat(src, span_danger("Hack aborted. [apc] is no longer responding to our systems."))
 		playsound(get_turf(src), 'sound/machines/buzz-sigh.ogg', 50, TRUE, ignore_walls = FALSE)
-	else
-		malf_picker.processing_time += 10
-
-		apc.malfai = parent || src
-		apc.malfhack = TRUE
-		apc.locked = TRUE
-		apc.coverlocked = TRUE
+		return
 
-		playsound(get_turf(src), 'sound/machines/ding.ogg', 50, TRUE, ignore_walls = FALSE)
-		to_chat(src, "Hack complete. [apc] is now under your exclusive control.")
-		apc.update_appearance()
+	malf_picker.processing_time += 10
+	var/area/apcarea = apc.area
+	if(istype(apcarea, /area/station/command/heads_quarters) || istype(apcarea, /area/station/ai_monitored/command/nuke_storage))
```

Comment:
I would replace this with an `is_type_in_list` and then make "`doomsday_bonus_apcs`" (or whatever you want to call the list of APCs which give you bonus points) either a global list or a static list on a relevant datum, so it's easy to mess with in the future.

Reply:
Added under the `/datum/ai_module/destructive/nuke_station` object as `var/list/discount_areas`.